docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,448 @@
+    res.send('Hi!');
+  });
+};
+```
+
+{% include code-caption.html content="src/components/hi.component.ts" %}
+
+```ts
+import {get} from '@loopback/rest';
+import {Component} from '@loopback/core';
+
+export class HiController {
+  @get('/hi')
+  hello(): string {
+    return 'Hi!';
+  }
+}
+
+export class HiComponent implements Component {
+  controllers = [HiController];
+}
+```
+
+{% include code-caption.html content="src/application.ts" %}
+
+```ts
+import { HiComponent} from './components/hi.component';
+...
+export class Lb4AppApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    ...
+    this.component(HiComponent);
+    ...
+  }
+}
+```
+
+Because of the architectural changes, `component-config.json` is not required
+in LoopBack 4 anymore.
+
+#### Boot scripts
+
+If you used LoopBack 3 boot scripts for adding routes to the app, it
+should now be moved to a standalone controller, a component, or implemented
+using `app.mountExpressRouter()`.
+
+Here is an example of migrating a route added by a boot script to a controller.
+
+{% include code-caption.html content="server/boot/hello.js" %}
+
+```js
+module.exports = function(server) {
+  server.get('/hello', (req, res) => {
+    res.send('Hello!');
+  });
+};
+```
+
+{% include code-caption.html content="src/controllers/hello.ts" %}
+
+```ts
+import {get} from '@loopback/rest';
+
+export class HelloController {
+  @get('/hello')
+  hello(): string {
+    return 'Hello!';
+  }
+}
+```
+
+{% include tip.html content="For details about migrating LoopBack 3 boot scripts
+refer to
+[Migrating boot scripts](https://loopback.io/doc/en/lb4/migration-boot-scripts.html)
+." %}
+
+#### Remote methods
+
+[Remote methods](https://loopback.io/doc/en/lb3/Remote-methods.html) add custom
+endpoints to a model's REST interface.
+
+{% include code-caption.html content="server/models/person.js" %}
+
+```js
+module.exports = function(Person) {
+  Person.greet = function(message, cb) {
+    cb(null, 'Greetings: ' + message);
+  }
+
+  Person.remoteMethod('greet', {
+    accepts: {arg: 'message', type: 'string'},
+    returns: {arg: 'message', type: 'string'}
+  });
+};
+```
+
+The above remote method's functionality can be migrated to a controller in
+LoopBack 4 in the following manner.
+
+{% include code-caption.html content="src/controllers/person.ts" %}
+
+```ts
+import {post, requestBody} from '@loopback/rest';
+
+export type Greeting = {
+  message: string;
+}
+
+const spec = {
+  content: {'application/json': {schema: {
+    type: 'object',
+    properties: {
+      message: {
+        type: 'string'
+      }
+    }
+  }}}
+};
+
+export class PersonController {
+  @post('/person/greet', {
+    responses: {
+      '200': spec,
+    },
+  })
+  greet(@requestBody(spec) greeting: Greeting): string {
+    return 'Greetings: ' + greeting.message;
+  }
+}
+```
+
+### Request/response pathway
```

Comment:
For this section, is it possible make the LoopBack 3 part shorter and introduce more differences between them? For example, we can pick a common use case in LB3, and show how we would do with LB4 path. Feel free to ignore if it's hard to achieve 😄

Reply:
Everything on this doc should be driven by LB3, as it is a migration guide. I tried my best to include all req-res related stuff from LB3 and show how it would be done in LB4. Let me know if I missed something.